### ISSUE TITLE
Sigma during training is different between the paper and this implementation

### ISSUE THREAD

**Issue by dhgrs** [NVIDIA/waveglow#17]
In the parper 2.1, you say that using sigma=sqrt(0.5). But in this implementation, sigma=1.
https://github.com/NVIDIA/waveglow/blob/6d260078e80963573d455c4785314f55eb6897b7/config.json#L6
Which one is used to train pretrained model?

**Comment by azraelkuan** [comment]
Also, the paper said that learning rare will be reduced from 1e-4 to 5e-5, but the code does not

**Comment by azraelkuan** [comment]
And when computer loss, i think that the logdet and log_s should not divided by z's shape, right?

**Comment by azraelkuan** [comment]
@dhgrs No, that is just the B and T, why not just 
```
log_s_total = log_s_total + torch.mean(log_s) 
log_det_W =  torch.logdet(W) / W.shape
z = torch.mean(z*z / (2*sigma *sigma)) + log_s_total + log_det_W
```

indeed, the final `z` shape is `(b x n_of_groups x 8)` and `8` is not the shape of `s` and `W`

**Comment by dhgrs** [comment]
@azraelkuan 
Because of early output, you can not do that.
For example, at 6th layer, `z`'s shape is `(b x n_of_groups x 6)` so that `torch.mean` divides with wrong shape. Then they sums at first.

**Comment by azraelkuan** [comment]
@dhgrs i know that. what i mean is in each flow, we compute the mean value first, and sum up them at last.

**Comment by dhgrs** [comment]
@azraelkuan 
The results between sum-first and mean-first are different. We should divide with `b x n_of_groups x 8` if it is at 6th layer and the shape is `(b x n_of_groups x 6)`.
If we mean-farst and sum-last, the mean of log|det W| gets larger. It means the loss gets smaller.

Am I misunderstanding?

**Comment by RPrenger** [comment]
@dhgrs Yes we used sigma=sqrt(0.5) in the paper, and the default training here is 1.0.  We used sqrt(0.5) as an accident of our implementation.  1.0 is a more logical default, and we highly doubt it'll make a difference to the model performance.

@azraelkuan Yes there is no learning rate schedule in the code.  We just hand changed the value once after a restart because it looked like training had leveled off.
